app/src/utils/api.ts

```diff
@@ -216,21 +215,28 @@ export const apiPOSTCall = async (apiUrl: string, postData: any) => {
 
     if (response.status === 200) {
       const data = await response.json();
-      return data?.nyplAPI?.response;
+      return data;
     } else {
-      return undefined;
+      throw new Error(
+        `${response.status}: ${response.statusText || "3xx/4xx error"}`
+      );
     }
   } catch (error) {
-    return undefined;
+    console.error(error);
+    throw new Error(error.message);
   }
 };
 
 export const getDivisionData = async (
-  slug: string,
+  slug?: string,
```

Comment:
Switch this around so that the optional argument is last in the list - general optional argument convention. But also, since you're giving them all defaults, they are all optional. This is called in `app/divisions/page.tsx` without any arguments.

Another pattern can also be to have these arguments passed as an object, such as:

```
const data = await getDivisionData({ slug: params.slug, pageNum: searchParams.page });
```

You'd still need the defaults and then the consumer can put the args in any order.